loopd/swapclient_server.go

```diff
@@ -165,11 +165,17 @@ func (s *swapClientServer) Monitor(in *looprpc.MonitorRequest,
 	// prevent subscribers from receiving duplicate updates.
 	swapsLock.Lock()
 
+	swaps, err := s.impl.FetchSwaps()
```

Comment:
I don't think this fixes it, because updating the swap in the database isn't happening inside the lock. Just sending out the notifications is in.

Reply:
Yes, I reverted the whole thing, see last comment.